cmd/drone-server/drone.go

```diff
@@ -236,6 +244,14 @@ func main() {
 		auth.POST("", server.GetLogin)
 	}
 
+	redirects := r.Group("/redirect")
+	{
+		redirects.Use(server.SetDatastore(store))
+		redirects.Use(server.SetRepo())
+
+		redirects.GET("/:owner/:name/commit/:sha", server.RedirectSha)
```

Comment:
Relying on sha or pull request number does have some consequences. This makes eventually replacing sqlite with Bolt a bit more complicated since we would need to manually maintain indexes of commit sha and pull request numbers.

I would like to explore all other options if possible. Although it sounds like there may be no other options :(